app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -591,13 +603,38 @@ export default class Heatmap {
 
   sortColumns(direction) {
     this.columnClustering = null;
+    this.gColumnDendogram.classed(cs.shouldSortColumns, true);
     orderBy(this.columnLabels, label => label.label, direction).forEach(
       (label, idx) => {
         label.pos = idx;
       }
     );
   }
 
+  sortRows(direction) {
+    // TODO (gdingle): sort unclassifieds to the bottom always
+    this.rowClustering = null;
+    this.gRowDendogram.classed(cs.shouldSortRows, true);
+    this.rowLabels = orderBy(
+      this.rowLabels,
+      label => label.sortKey || label.label,
+      direction
+    );
+    this.rowLabels.forEach((label, idx) => {
+      label.pos = idx;
+    });
+
+    // Need to sort this array as well. It is created from rowLabels earlier.
+    this.filteredRowLabels = orderBy(
```

Comment:
Isn't this the only thing that is shown and thus the only one you actually need to sort?

Reply:
That's correct, but I thought it was important to also sort the `rowLabels` because that variable is used elsewhere and it would be surprising if it was in a different order.

Reply:
Alternatively, naming it `filteredAndSortedRowLabels` would make the intentions clearer.

Reply:
Yeah, if rowLabels is used somewhere else it would be important. I guess I find it weird if it is, since they do not match... Do you know what it is used for?